packages/web/src/new/sdk/SolanaSDK/models/APIEndpoint.ts

```diff
@@ -1,36 +1,49 @@
 import type { Network } from '@saberhq/solana-contrib';
 import { clusterApiUrl } from '@solana/web3.js';
 
+import { RemoteConfig } from 'new/services/RemoteConfig';
+
+enum APIKeysNames {
+  rpcpool = 'rpcpool',
+}
+
+const API_KEYS: Record<APIKeysNames, string> = {
+  rpcpool: process.env.REACT_APP_RPCPOOL_API_KEY as string,
+};
+
+export type APIEndpointProps = {
+  address: string;
+  network: Network;
+  socketUrl?: string;
+  additionalQuery?: string;
+};
+
 export class APIEndpoint {
   address: string;
   network: Network;
   socketUrl: string;
   additionalQuery?: string;
 
-  constructor({
-    address,
-    network,
-    socketUrl,
-    additionalQuery,
-  }: {
-    address: string;
-    network: Network;
-    socketUrl?: string;
-    additionalQuery?: string;
-  }) {
+  private _apiKeyName?: APIKeysNames;
+
+  constructor({ address, network, socketUrl, additionalQuery }: APIEndpointProps) {
     this.address = address;
     this.network = network;
     this.socketUrl = socketUrl ?? address.replace('http', 'ws');
-    this.additionalQuery = additionalQuery;
+
+    const apiKeyName = additionalQuery as APIKeysNames;
+    this.additionalQuery = API_KEYS[apiKeyName] || additionalQuery;
+
+    this._apiKeyName = API_KEYS[apiKeyName] ? apiKeyName : undefined;
```

Comment:
don't see place where its using

Reply:
it's used in toJSON to replace api key by constant name. When we save Defaults.apiEndpoint to LocalStorage. If api key constant name was passed in additionalQuery. Otherwise value of additionalQuery is used

https://github.com/p2p-org/p2p-wallet-web/blob/b24d5ecfbe80d2d9199b114fd68ad86f17ba6876/packages/web/src/new/sdk/SolanaSDK/models/APIEndpoint.ts#L105